src/plugins/list.js

```diff
@@ -11,7 +13,34 @@ function List() {
    * @param {Editor} editor
    * @param {Function} next
    */
-  const onEnter = (event, editor, next) => next();
+  const onEnter = (event, editor, next) => {
+    const { value } = editor;
+    const { startBlock } = value;
+    event.preventDefault();
+
+    // Hitting enter on a blank list item will break out of the enclosing list
+    if (isSelectionInput(value, CONST.LIST_ITEM) && startBlock.text.length === 0) {
+      editor.withoutNormalizing(() => {
+        event.preventDefault();
+        editor
+          .setBlocks(CONST.PARAGRAPH)
+          .unwrapBlock(CONST.LIST_ITEM)
+          .unwrapBlock(currentList(value).type);
+      });
+      return false;
+    }
+
+    // Hitting enter on a non-empty list item will add a new list_item
+    if (isSelectionInput(value, CONST.LIST_ITEM) && startBlock.text.length !== 0) {
+      editor.withoutNormalizing(() => {
+        event.preventDefault();
+        editor.splitBlock().unwrapBlock(CONST.LIST_ITEM).wrapBlock(CONST.LIST_ITEM);
```

Comment:
What is .unwrapBlock(CONST.LIST_ITEM).wrapBlock(CONST.LIST_ITEM) doing on line 37? Why would we unwrap a list item just to wrap it again?

Reply:
I made this diagram that hopefully helps explain the process.

![Image from iOS (1)](https://user-images.githubusercontent.com/36460856/69085173-9e9d8080-0a13-11ea-908d-4cae9081cc41.jpg)

Reply:
Thanks for explaining this. Makes sense now!